lib/active_merchant/billing/gateways/hps.rb

```diff
@@ -47,15 +47,25 @@ def capture(money, transaction_id, options={})
         end
       end
 
-      def purchase(money, card_or_token, options={})
-        commit('CreditSale') do |xml|
-          add_amount(xml, money)
-          add_allow_dup(xml)
-          add_customer_data(xml, card_or_token, options)
-          add_details(xml, options)
-          add_descriptor_name(xml, options)
-          add_payment(xml, card_or_token, options)
-          add_three_d_secure(xml, card_or_token, options)
+      def purchase(money, payment_method, options={})
+        if payment_method.is_a?(Check)
+          commit('CheckSale') do |xml|
+            add_check_payment(xml, payment_method, options)
+            add_amount(xml, money)
+            xml.hps :SECCode, 'WEB'
```

Comment:
Thinking of making this a GSF, since we already need to implement one for `check_void`.

Reply:
Also thinking about making it a helper method for uniformity, but this is literally all that will ever go in it, so I'm 50/50 on that.

Reply:
I agree with being 50/50 on making this a helper method. There are a good number of single line methods in this gateway, though, so that might tip the scales sufficiently into making it into a method.

I do think that it would be worthwhile to make this a GSF. I can certainly see a use case for specifying telephone orders.